### DISCUSSION TITLE
support for custom anchors in headers

### DISCUSSION THREAD

**Discussion by meeusr** [mkdocs/mkdocs#3543] [Q&A]
Hello

I'm trying to use a custom anchor for the header using the following syntax:
```
# chapter 2 { #chapter-two }
```
but is looks like it is not accepted since the rendered html shows:
```
chapter 2 { #chapter-two }
```
which links to .../chapter2.html#chapter-2-chapter-two.

For code-blocks the assignment of IDs and the linking seems to work well.
Am I missing something or do I use it in the wrong way?

Background is: I would like to use custom tags instead of auto-generated one's based on the header text because I think it is more user friendly these sections are linked from a lot of other documents.

Thanks

Ronny

**Comment by Andre601** [comment]
You need the attr_list extension to enable this

**Reply by meeusr** [reply]
I tested it and it indeed works. 
Thanks for the quick feedback.

Would it be a good approach to use these custom tags or is there a better solution?

**Reply by Andre601** [reply]
I think it's the best solution there is
